cmd/cmd_test.go

```diff
@@ -0,0 +1,51 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+const cfgHomeKey = "EXERCISM_CONFIG_HOME"
+
+type CommandTest struct {
+	App            *cobra.Command
+	Cmd            *cobra.Command
+	InitFn         func()
+	TmpDir         string
+	Args           []string
+	OriginalValues struct {
+		ConfigHome string
+		Args       []string
+	}
+}
+
+func (test *CommandTest) Setup(t *testing.T) {
```

Comment:
Both `Setup` and `Teardown` take a `testing.T`, which means that they can `t.Fatal` if things go terribly wrong, or `assert.NoError` for things that might go a little bit wrong.